src/arrays/arrays_with_methods/hard/q7/MainTest.java

```diff
@@ -0,0 +1,31 @@
+package arrays.arrays_with_methods.hard.q7;
+
+import global.tools.CustomAssertions;
+import global.utils.MethodUtil;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest {
+    // Java
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(
```

Comment:
You should use arrayUtil to generate some very large test cases (like 500 plus).